routing/payment_session.go

```diff
@@ -257,6 +265,7 @@ func (p *paymentSession) RequestRoute(maxAmt, feeLimit lnwire.MilliSatoshi,
 		CltvLimit:          cltvLimit,
 		DestCustomRecords:  p.payment.DestCustomRecords,
 		DestFeatures:       p.payment.DestFeatures,
+		HopFeatures:        p.payment.HopFeatures,
```

Comment:
NOTE: Not sure if it makes most sense to introduce the pathfinding restriction here in addition to the _**QueryRoutes**_ flow. This may require exposing in the _**SendPaymentV2**_ --> _**extractIntentFromSendRequest**_, **_<routing.ChannelRouter>.SendPaymentAsync_** --> **_sendPayment_** --> **_resumePayment_** --> **_requestRoute_** -->  **_\<paymentSession\>.RequestRoute_** flow.

Come to think of it, I'm not even sure whether the _**QueryRoutes**_ RPC is the right place to expose restricting intermediate hop features during path finding - at least as far as blinded routes go, since I am not sure what the API for selecting introduction nodes and building the route will actually look like.